frontends/precinct-scanner/src/app.test.tsx

```diff
@@ -99,17 +93,35 @@ const getMarkThresholdOverridesConfigNoMarkThresholdOverridesResponseBody: Scan.
     status: 'ok',
   };
 
-const getMarkThresholdOverridesConfigResponseBody: Scan.GetMarkThresholdOverridesConfigResponse =
-  {
-    status: 'ok',
-    markThresholdOverrides: { definite: 0.5, marginal: 0.25 },
-  };
+let card: MemoryCard;
+let storage: MemoryStorage;
+let hardware: MemoryHardware;
+let kiosk = fakeKiosk();
 
-test('shows setup card reader screen when there is no card reader', async () => {
-  const card = new MemoryCard();
-  const storage = new MemoryStorage();
-  const hardware = MemoryHardware.buildStandard();
-  hardware.setCardReaderConnected(false);
+const pollWorkerCard = makePollWorkerCard(
+  electionSampleDefinition.electionHash
+);
+
+const electionManagerCard = makeElectionManagerCard(
+  electionSampleDefinition.electionHash,
+  '123456'
+);
+
+beforeEach(() => {
+  jest.useFakeTimers();
+
+  card = new MemoryCard();
+  storage = new MemoryStorage();
+  hardware = MemoryHardware.build({
+    connectCardReader: true,
+    connectPrecinctScanner: true,
+  });
```

Comment:
Loved that you factored out all this repeated code!

In general I try to avoid having global shared objects between tests, because I find that sometimes tests can get intertwined and mess with each other in weird ways. Seems like this is working fine right now, so you don't necessarily need to change it, but I lean towards a pattern of trying to keep as much local to a test function as possible.

I factoring out repeated setup functionality by creating a function to do that work and return the results, e.g.:

```ts
function setupFixtures() {
  return {
      card: new MemoryCard(),
      storage: new MemoryStorage(),
      ... etc ...
  }
}
```
and then I destructure the result in each test:
```ts
const { card, storage } = setupFixtures()
```

Or sometimes I even include the rendering of the component in that setup function!

Reply:
I see, that seems like a safer pattern. I'm going to leave it for now but in the future plan to do what you suggest.